### PR TITLE
Import Konsole 22.08.0

### PR DESCRIPTION
**Pull request by travier**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub: Member of the @flathub/kde team
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by travier** [comment]
On Fedora 36 KDE Wayland, the app opens for a spit second and then terminates. Nothing in the command line log.

**Comment by travier** [comment]
(CC @flathub/kde for the general app submission)

**Comment by travier** [comment]
With `flatpak run --user org.kde.konsole --noclose`:
```
/bin/bash: /bin/bash: cannot execute binary file
```

**Comment by travier** [comment]
CC @hsitter

**Comment by Mailaender** [comment]
I assume you have to bundle `bash`.

**Comment by hsitter** [comment]
@travier if you could try the patch from https://invent.kde.org/utilities/konsole/-/merge_requests/728 that would be amazing.

**Comment by travier** [comment]
> I assume you have to bundle `bash`.

We already do: you can try with `flatpak run --command=bash foo`

**Comment by travier** [comment]
> @travier if you could try the patch from [invent.kde.org/utilities/konsole/-/merge_requests/728](https://invent.kde.org/utilities/konsole/-/merge_requests/728) that would be amazing.

Thanks, will push it here.

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by travier** [comment]
OK, this fixes the issue and now I get an interactive shell. Thanks @hsitter!
However, I can not run true terminal commands such as `htop`:
```
$ htop
Error opening terminal: unknown.
```

Outside:
```
$ echo $TTY
/dev/pts/2
```

Inside:
```
$ echo $TTY

```

Maybe we need something like what's done in https://github.com/1player/host-spawn instead of using flatpak-spawn.

**Comment by travier** [comment]
We might need a way to change https://invent.kde.org/frameworks/kcoreaddons/-/blob/master/src/lib/util/ksandbox.cpp#L49 without having to recompile the whole Runtime.

**Comment by hsitter** [comment]
For testing purposes we could just build kcoreaddons as part of the konsole manifest and see where that takes us. If host-spawn works better for us I can look into giving us more flexibility in ksandbox

**Comment by TheEvilSkeleton** [comment]
Interestingly, when launching, I get this error when using fish by default:
```
Warning: Could not find '/bin/fish', starting '/bin/sh' instead.  Please check your profile settings.
```

However, `/bin/fish` does indeed exist:
```
sh-5.1$ ls /bin/fish
/bin/fish
```

Running `fish` outputs this:
```
sh-5.1$ /bin/fish
warning: Could not set up terminal.
warning: TERM environment variable not set.
```

**Comment by hfiguiere** [comment]
Shouldn't the terminal just open a pty that will actually run everything on the host outside the sandbox?

**Comment by hsitter** [comment]
Curiously when I open konsole from the menu htop works as expected, meanwhile when started from a host tty it is broken. Maybe this problem actually has to do with the fact that we can't have a controlling tty (ctty) inside the sandbox and so the ctty ends up being the host konsole and thus things are getting confused?

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by tsdgeos** [comment]
I still get nothing with
 flatpak run org.kde.konsole

**Comment by travier** [comment]
Sorry, the last push is broken. I've try with host-spawn and I get better results. I'll work on a patch for ksandbox to use that as an option.

**Comment by travier** [comment]
I'm currently fighting to build the Go binary from source in the Flatpak.

This one from @hsitter has good results: https://github.com/hsitter/flathub/tree/org.kde.konsole.
Weirdly, running `host-spawn` directly from `flatpak run --command=bash org.kde.konsole` works well with resizes, but running it in Konsole does not work well.

**Comment by travier** [comment]
We could figure out some scheme to pass env variables (https://github.com/1player/host-spawn/blob/master/main.go#L74).

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by travier** [comment]
Latest push does not work for resizes but spawns a shell at least.

**Comment by 1bcb** [comment]
The newest build works for me in Mint 21 Cinnamon.

**Comment by Newbytee** [comment]
bot, build org.kde.konsole

**Comment by 1bcb** [comment]
This one also works for me on Mint 21 Cinnamon.

**Comment by Newbytee** [comment]
> This one also works for me on Mint 21 Cinnamon.

Yeah, it should be identical. I just rebuilt it because the previous one had expired and I wanted to test it.

**Comment by travier** [comment]
Things should be a little bit better now. We have two remaining known issues:
* crashes on right click
* wrong check for binary for non custom profiles

**Comment by travier** [comment]
https://invent.kde.org/utilities/konsole/-/merge_requests/747

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by TheEvilSkeleton** [comment]
bot, build org.kde.konsole

**Comment by travier** [comment]
See https://github.com/flathub/flathub/pull/3447#issuecomment-1264652401 & https://github.com/flathub/flathub/pull/3447#issuecomment-1264693778 which should be resolved before me merge this one.

**Comment by Waqar144** [comment]
Merge requests to fix two issues the one mentioned above:

- https://invent.kde.org/utilities/konsole/-/merge_requests/829
- https://invent.kde.org/utilities/konsole/-/merge_requests/829

Hoping to get both of these into 23.04

**Comment by travier** [comment]
Thanks!

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by travier** [comment]
Looks great to me! Thanks @Waqar144 !

**Comment by travier** [comment]
There is a one remaining issue but it's not critical: Closing a tab will ask you to confirm as Konsole thinks that there is still a program running.

File: org.kde.konsole.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    "runtime-version": "5.15-22.08",
+    "sdk": "org.kde.Sdk",
+    "command": "konsole",
+    "rename-icon": "utilities-terminal",
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc",
+        "--socket=wayland",
+        "--socket=fallback-x11",
+        "--talk-name=org.a11y.Bus",
```
Comment (Erick555) [inline org.kde.konsole.json:13]: This should be removed, see https://github.com/flathub/org.kde.ark/commit/ec438fb43c7ec1f71b0c946f5c54232723a7f8ba

**Comment by Waqar144** [comment]
> There is a one remaining issue but it's not critical: Closing a tab will pop a confirmation dialog as Konsole thinks that there is still a program running.

Simply avoiding the popup is easy but that doesn't fix anything and can lead to the user accidentally closing  a tab when a process was running.

Konsole needs to know about any foreground running processes. This doesn't seem possible in a direct way right now.

File: org.kde.konsole.json (hunk 1 above)
Comment (travier) [reply org.kde.konsole.json:13]: Good catch

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by Waqar144** [comment]
Also, not sure yet as I haven't tested but we might need `filesystem=host:ro` `filesystem=/usr/include:ro`. Konsole has a feature where it scans the text for possible links to files and then if you Ctrl+Click on a file it can open it for you using the default program.

**Comment by hfiguiere** [comment]
`filesystem=/usr/include:ro` this won't work

**Comment by Erick555** [comment]
The `:ro` would also make everything unwritable unless sandbox is escaped through flatpak-spawn.

**Comment by Waqar144** [comment]
> The :ro would also make everything unwritable ,,

Write access is not needed as far as I understand / know.

The reason it needs access is because it regex matches the scrollback history and looks for files that match the current directory's files and allows you to access them. For relative paths this is completely broken atm.

For absolute it does highlight the links, but isn't able to do much. Probably a bug somewhere else in the KDE frameworks. Konsole doesn't do much checking for absolute paths.

**Comment by Erick555** [comment]
> Write access is not needed as far as I understand / know.
For this feature or at all? There won't be write access at all.

**Comment by travier** [comment]
We're already running everything outside the sandbox via flatpak-spawn. Konsole should check if files exists that way. I don't think we should add more holes in the sandbox as using two different mechanisms to look for things will definitely be confusing.

While I agree that this is a bug, I don't think it should stop including it in Flathub for now.

**Comment by Waqar144** [comment]
Yes, its not a blocker.

**Comment by travier** [comment]
> Konsole has file backed scrollback history which is stored somewhere (not sure where atm). That might need write access to a specific directory. Maybe its stored in `/tmp`, will have to check

That should stay in the sandbox.

**Comment by travier** [comment]
bot, build org.kde.konsole

**Comment by travier** [comment]
This is working well for me. Should be ready now.

**Comment by tsdgeos** [comment]
are we aiming beta or "normal"?

**Comment by travier** [comment]
I think we still have https://github.com/flathub/flathub/pull/3447#issuecomment-1469803713 (not re-checked), so maybe Beta?

**Comment by tsdgeos** [comment]
I'm good with beta :)

**Comment by TheEvilSkeleton** [comment]
> I think we still have [#3447 (comment)](https://github.com/flathub/flathub/pull/3447#issuecomment-1469803713) (not re-checked), so maybe Beta?

I tested on my end and I can reproduce it

**Comment by Waqar144** [comment]
Keeping it in beta is okay, but that generally means close to 0 real users. IMO, konsole is fairly usable now so we should aim for stable so that more people are able to use it and test it.

**Comment by travier** [comment]
The remaining issue is not critical I agree. I does not break the experience, just makes it less fluid. I'm OK with stable or beta, both will be better than keeping it here

**Comment by travier** [comment]
Let's merge it in stable and track the issue? @tsdgeos

**Comment by travier** [comment]
Thanks!

**Pull request closed without merging**